packages/cloudflare/src/index.ts

```diff
@@ -84,6 +89,59 @@ export default function createIntegration(args?: Options): AstroIntegration {
 							wasmModuleLoader({
 								disabled: !args?.wasmModuleImports,
 							}),
+							chunkAnalyzer.getPlugin(),
+							{
+								name: 'prerender-scanner',
+								enforce: 'post',
+								generateBundle(_, bundle) {
+									// Loop through all chunks and find out which pages are prerendered
+									for (const chunk of Object.values(bundle)) {
+										if (chunk.type !== 'chunk') continue;
+										if (
+											chunk.dynamicImports.some(
+												(entry) =>
+													entry.includes('prerender') &&
+													chunk.name !== '_@astrojs-ssr-virtual-entry'
+											)
+										) {
+											prerenderImports.push([chunk.facadeModuleId ?? '', chunk.fileName]);
+										}
+									}
+
+									const entryChunk = bundle['index.js'];
+									if (
+										entryChunk &&
+										entryChunk.type === 'chunk' &&
+										entryChunk.name === '_@astrojs-ssr-virtual-entry'
+									) {
+										entryChunk.dynamicImports = entryChunk.dynamicImports.filter(
+											(entry) => !prerenderImports.map((e) => e[1]).includes(entry)
+										);
+										for (const page of prerenderImports) {
+											const importRegex = new RegExp(
```

Comment:
I think we should add a comment that points to the actual place where Astro creates/writes these modules, so we know why it's there and how to fix it in case upstream decides to change things, even the name of a page.